### ISSUE TITLE
Date.daysBetween return wrong result

### ISSUE THREAD

**Issue by Askolein** [amaggiulli/qlnet#106]
Hi all, 

In the [file Date.cs](https://github.com/amaggiulli/qlnet/blob/master/QLNet/Time/Date.cs) at line 68, the current implementation of static method `daysBetweenDate` is
```
public static double daysBetween(Date d1, Date d2) 
{
    double days = d2 - d1;
    return days + d2.fractionOfDay() - d1.fractionOfDay();
}
```

In my opinion the correct implementation should be:
```
public static double daysBetween(Date d1, Date d2) 
{
    return (d2.date - d1.date).TotalDays;
}
```

To illustrate the error of the current implementation, here is a correction in the current implementation (same result as the one proposed above):
```
public static double daysBetween(Date d1, Date d2) 
{
    double days = d2 - d1;
    double fraction = d2.fractionOfDay() - d1.fractionOfDay();
    if (fraction < 0.0)
        days++;
			
    return days + d2.fractionOfDay() - d1.fractionOfDay();
}
```
	
Otherwise it won't work for dates where d2 time part is earlier in the day than d1 time part. A good example is:

	d1 = new Date(new DateTime(2016, 1, 1, 18, 0, 0));
	d2 = new Date(new DateTime(2016, 1, 2, 0, 0, 0));

**Expected result is: 0.25 but we currently get -0.75**

What do you think?

Cheers.

**Comment by igitur** [comment]
The error is actually in the operator overload:
```
public static int operator -(Date d1, Date d2) { return (d1.date- d2.date).Days; }
```
should be:
```
public static int operator -(Date d1, Date d2) { return (d1.date.Date - d2.date.Date).Days; }
```

That fixes it.

But I think your fix (using `TotalDays`) will improve performance even more.

**Comment by Askolein** [comment]
Well spotted @igitur . 
Best approach would be to use `TotalDays` for the `daysBetween(..)`method but also to fix the operator as you suggested.

These are major bugs that impact most of the results. Can it be fixed? Included in the next release? 

Thanks.

**Comment by amaggiulli** [comment]
Well spotted ,  fixed in cc4ef00defa92555df38c03facbc29bfb7f577c9 . Will be in the next major release
